### PR TITLE
Add Security Considerations section on deletion.

### PR DESCRIPTION
**Pull request by msporny**
This PR is an attempt to address issue #276 by adding a Security Considerations section on deletion.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/360.html" title="Last updated on Jan 23, 2024, 7:20 PM UTC (2236468)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/360/6a4579c...2236468.html" title="Last updated on Jan 23, 2024, 7:20 PM UTC (2236468)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1017,6 +1017,51 @@ <h2>Privacy Considerations</h2>
...
     </p>
+
+    <section>
+      <h3>Deletion</h3>
+
+      <p>
+The APIs provided by this specification enable the deletion of
+<a>verifiable credentials</a> and <a>verifiable presentations</a> from
+<a href="#storage-services">storage services</a>. The result of these deletions
+and the side-effects caused by them are out of scope for this specification.
```
Comment (TallTed) [inline index.html:1028]:
```suggestion
and the side-effects they might cause are out of scope for this specification.
```

File: index.html [hunk 2]
```diff
@@ -1017,6 +1017,51 @@ <h2>Privacy Considerations</h2>
...
+implemented. There are at least two types of deletion that are contemplated by
+this specification.
+      </p>
+
+      <p>
+<dfn>Partial deletion</dfn> marks a record for deletion but continues to store
+some or all of the original information. This mode of operation can be useful if
+there are requirements to be able to audit all credentials or presentations over
+a particular time period or if recovering the original credential might be a
+useful feature to provide an entity.
```
Comment (TallTed) [inline index.html:1039]:
```suggestion
some or all of the original information. This mode of operation can be useful if
there are audit requirements for all credentials and/or presentations over
a particular time period, or if recovering an original credential might be a
useful feature to provide to an entity.
```

File: index.html [hunk 3]
```diff
@@ -1017,6 +1017,51 @@ <h2>Privacy Considerations</h2>
...
+      </p>
+      <p>
+<dfn>Complete deletion</dfn> purges all information related to a given
+<a>verifiable credential</a> or <a>verifiable presentation</a> in a way that
+is unrecoverable. This mode of operation can be useful when removing information
+that is outdated and beyond the needs of an audit or when responding to any
```
Comment (TallTed) [inline index.html:1045]:
```suggestion
that is outdated and beyond the needs of any audit or when responding to any
```

File: index.html [hunk 4]
```diff
@@ -1017,6 +1017,51 @@ <h2>Privacy Considerations</h2>
...
+sort of "<a href="https://en.wikipedia.org/wiki/Right_to_be_forgotten">right
+to be forgotten</a>" request.
+      </p>
+      <p>
+When deleting a <a>verifiable credential</a>, what to do with the status
+information needs to be considered. Some use cases might call for a deletion
+of a particular <a>verifiable credential</a> to also set the revocation
+and suspension bits of the <a>verifiable credential</a> such that any sort of
+status check for the deleted credential fails and usage of the credential is
+halted.
```
Comment (TallTed) [inline index.html:1055]:
```suggestion
When deleting a <a>verifiable credential</a>, handling of its status
information needs to be considered. Some use cases might call for deletion
of a particular <a>verifiable credential</a> to also set the revocation
and suspension bits of that <a>verifiable credential</a>, such that any sort of
status check for the deleted credential fails and use of the credential is
halted.
```

File: index.html [hunk 5]
```diff
@@ -1017,6 +1017,51 @@ <h2>Privacy Considerations</h2>
...
+      </p>
+      <p>
+Given the scenarios above, it is advisable that implementers allow the
+moderation of what happens after a delete to be programmable such that
+system flexibility is achieved to address any <a>verifiable credential</a> use
+case.
```
Comment (TallTed) [inline index.html:1061]:
```suggestion
Given the scenarios above, implementers are advised to allow the system actions
that occur after a delete to be configurable, such that system flexibility is
sufficient to address any <a>verifiable credential</a> use case.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
minor, human, editorial

**Review by brianorwhatever — APPROVED** [review]

**Review by dlongley — APPROVED** [review]
+1 w/ @TallTed's suggestions.

**Pull request merged**